pkg/download/addons/with_pool.go

```diff
@@ -127,3 +129,21 @@ func (p *withpool) Zip(ctx context.Context, mod, ver string) (io.ReadCloser, err
 	}
 	return zip, nil
 }
+
+func (p *withpool) Catalog(ctx context.Context, token string, pageSize int) ([]paths.AllPathParams, string, error) {
+	const op errors.Op = "pool.Catalog"
+	var modsVers []paths.AllPathParams
+	var nextToken string
+	var err error
+	done := make(chan struct{}, 1)
+	p.jobCh <- func() {
+		modsVers, nextToken, err = p.dp.Catalog(ctx, token, pageSize)
+		close(done)
```

Comment:
minor: can be done with waitgroup

Reply:
I kept it consistent with all the other methods of withpool. They all have the same structure, I would not change that.